### PR TITLE
Fix embeddings settings update

### LINKED ISSUE #4585: Meilisearch 1.8.0-rc.1 stuck process when adding hugging face and ollama embedders
**Describe the bug**
When I test adding either hugging face or ollama embedders Meilisearch seems to be stuck processing indefinently on v1.8.0-rc1. Running the exact same code on v1.8.0-rc0 doesn't have this issue.

**To Reproduce**
Here is a minimum reproducable Python program

```py
import asyncio

from httpx import AsyncClient as HttpxAsyncClient

from meilisearch_python_sdk import AsyncClient
from meilisearch_python_sdk.models.settings import (
    HuggingFaceEmbedder,
    MeilisearchSettings,
)


async def enable_vector_search() -> None:
    async with HttpxAsyncClient(
        base_url="http://127.0.0.1:7700", headers={"Authorization": "Bearer masterKey"}
    ) as client:
        await client.patch("/experimental-features", json={"vectorStore": True})


async def main() -> int:
    await enable_vector_search()
    async with AsyncClient("http://127.0.0.1:7700", "masterKey") as client:
        settings = MeilisearchSettings(
            embedders={
                "default": HuggingFaceEmbedder(),  # bad
            },
        )
        await client.create_index("movies", settings=settings)

    return 0


if __name__ == "__main__":
    raise SystemExit(asyncio.run(main()))
```

**Expected behavior**
The process should either compete or I should at least get an error.

**Screenshots**

**Meilisearch version:**
v1.8.0-rc.1

**Additional context**
Additional information that may be relevant to the issue.
[e.g. architecture, device, OS, browser]

### PR DESCRIPTION
**Pull request by ManyTheFish**
# Pull Request
- add some conditions reducing the work done when changing the settings
- add some benchmarks on embedders

## Related issue
Fixes #4585

### CODE REVIEW

**Comment by ManyTheFish** [comment]
Building a prototype `prototype-embeddings-opti-0` to make the user test it

File: milli/src/update/index_documents/extract/extract_vector_points.rs [hunk 1]
```diff
@@ -198,11 +198,16 @@ pub fn extract_vector_points<R: io::Read + io::Seek>(
...
+                        // TODO: this filter works because we erase the vec database when a embedding setting changes.
+                        // When vector pipeline will be optimized, this should be removed.
+                        .filter(|_| !settings_diff.reindex_vectors())
+                        .map(|p| {
+                            p.render(obkv, DelAdd::Deletion, old_fields_ids_map).unwrap_or_default()
+                        });
                     let new_prompt = prompt.render(obkv, DelAdd::Addition, new_fields_ids_map)?;
-                    if old_prompt != new_prompt {
+                    if old_prompt.as_ref() != Some(&new_prompt) {
+                        let old_prompt = old_prompt.unwrap_or_default();
```
Comment (Kerollmops) [inline milli/src/update/index_documents/extract/extract_vector_points.rs:210]: Ho! So, the main slowness issue was that we were recomputing the embedding even if the prompt was the same, right?

**Review by Kerollmops — APPROVED** [review]
Thank you for the fix Many le mops 👏
Awaiting user approval once he confirms that the prototype works!

**Comment by ManyTheFish** [comment]
bors merge

**Pull request merged**